tutorials/define_and_compile_model.py

```diff
@@ -0,0 +1,72 @@
+"""
+Quick Start - Compile Resnet-18 Model for Raspberry Pi
+===========================================
+**Author**: `Yao Wang <https://github.com/kevinthesun>`_
+
+This example shows how to build a neural network with NNVM python frontend and generate runtime library for Raspberry Pi with TVM.
+To run this notebook, you need to install tvm and nnvm following `these instructions <https://github.com/dmlc/nnvm/blob/master/docs/how_to/install.md>`_.
+Notice that you need to build tvm with llvm.
+
+To begin with, we import NNVM and TVM.
+"""
+import tvm
+import nnvm.compiler
+import nnvm.testing
+
+
+######################################################################
+# Define Neural Network in NNVM
+# -----------------------------
+# First, let's define a neural network with nnvm python frontend.
+# For simplicity, we'll use pre-defined resnet-18 network in NNVM.
+# Parameters are initialized with Xavier initializer.
+# NNVM also supports other model formats such as MXNet, CoreML and ONNX.
+#
+# In this tutorial, we assume we will do inference on our device
+# and the batch size is set to be 1. Input images are RGB color
+# images of size 224 * 224. We can call the :any:`nnvm.symbol.debug_str`
+# to show the network structure.
+
+batch_size = 1
+image_shape = (3, 224, 224)
+data_shape = (batch_size,) + image_shape
+
+net, params = nnvm.testing.resnet.get_workload(batch_size=batch_size, image_shape=image_shape)
+print(net.debug_str())
+
+######################################################################
+# Compile with TVM
+# ----------------------------
+# Next step is to compile model with tvm.
+# The key parameter for compilation is optimization level.
+# Currently this value can be 0 to 2, which corresponds to
+# "SimplifyInference", "OpFusion" and "PrecomputePrune" respectively.
+# In this example we set optimization level to be 0
+# and use Raspberry Pi as compile target.
+#
+# By calling :any:`nnvm.compiler.build`, you will get three return values: graph,
+# library and the new parameter, since we do some optimization that will
+# change the parameters but keep the result of model same as original.
```

Comment:
`nnvm.compiler.build` returns three compoents: the execution graph in json format, the TVM module library of compiled functions specifically for this graph on the target hardware, and the parameter blobs of the model. During the compilation, NNVM does the graph-level optimization while TVM does the tensor-level optimization, resulting in an optimized runtime module for model serving.